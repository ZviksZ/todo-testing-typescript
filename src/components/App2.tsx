import * as React                                from 'react';
import Counter from "./Counter";

function App2() {
   return (
      <div className="App">
         <Counter />
      </div>
   );
}

export default App2;
