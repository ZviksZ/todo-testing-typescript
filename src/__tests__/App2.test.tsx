import * as React                                from 'react';
import { mount } from "enzyme";
import App2 from "../components/App2";

describe("Counter Testing", () => {
   let wrapper;
   beforeEach(() => {
      wrapper = mount(<App2 />);
   });
   test("render snapshot", () => {
      expect(wrapper).toMatchSnapshot();
   });
   test("render the title of counter", () => {
      expect(wrapper.find("h1").text()).toContain("This is counter app");
   });

   test("render a button with text of `increment`", () => {
      expect(wrapper.find("#increment-btn").text()).toBe("Increment");
   });

   test("render the initial value of state in a div", () => {
      expect(wrapper.find("#counter-value").text()).toBe("0");
   });

   test("render the click event of increment button and increment counter value", () => {
      wrapper.find("#increment-btn").simulate("click");
      expect(wrapper.find("#counter-value").text()).toBe("1");
   });

   test("render the click event of decrrment button and decrement counter value", () => {
      wrapper.find("#increment-btn").simulate("click");
      expect(wrapper.find("#counter-value").text()).toBe("1");
      wrapper.find("#decrement-btn").simulate("click");
      expect(wrapper.find("#counter-value").text()).toBe("0");
   });
});
