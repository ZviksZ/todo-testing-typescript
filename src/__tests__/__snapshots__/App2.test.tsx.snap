// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Testing render snapshot 1`] = `
<App2>
  <div
    className="App"
  >
    <Counter>
      <div>
        <h1>
          This is counter app
        </h1>
        <div
          id="counter-value"
        >
          0
        </div>
        <button
          id="increment-btn"
          onClick={[Function]}
        >
          Increment
        </button>
        <button
          id="decrement-btn"
          onClick={[Function]}
        >
          Decrement
        </button>
      </div>
    </Counter>
  </div>
</App2>
`;
